### ISSUE TITLE
Consider adding API Gateway support

### ISSUE THREAD

**Issue by nyilmaz** [localytics/lambda-slack-router#33]
If lambda function is behind the API Gateway then returned response should be stringified in body parameter as such;

```
// build a response to return to Slack
var buildResponse = function (response_type, response) {
  var modifiedResponse = response;
  if (typeof response === 'string') {
    return { response_type: response_type, text: response };
  }
  modifiedResponse.response_type = response_type;
  return {
  	body: JSON.stringify(modifiedResponse);
  }
};
```

Since gateway forwards the body content to Slack. Maybe this feature would be controlled by a simple boolean parameter to `inChannelResponse` and `ephemeralResponse ` functions and `buildResponse` function can modify returned object accordingly.

**Comment by kddeisz** [comment]
Interesting, I wonder what our configuration is that's different. It's working for us behind API gateway (we're using [serverless](https://serverless.com/) to deploy).

**Comment by kddeisz** [comment]
@nyilmaz are you unable to get this to work as it currently stands?

**Comment by nyilmaz** [comment]
Unfortunately yes. I thought that API does not forward response object returned from `buildResponse`.

**Comment by kddeisz** [comment]
Hmm I wonder if this line is what's fixing it for us: https://github.com/localytics/serverless-slackbot-scaffold/blob/9dfaadcbb6e5a087af46e146637aca1cdb82cbbc/template/slackbot/s-function.json#L23. That's from our slackbot scaffold. Does that fix it for you?

**Comment by nyilmaz** [comment]
Yep, probably that is the line transforms the response. We are using API gateway as it is without any request templates. It's just proxy for us.

**Comment by kddeisz** [comment]
Makes sense. This should solve your problem: #34

**Comment by nyilmaz** [comment]
Thanks, @kddeisz !

**Comment by kddeisz** [comment]
@nyilmaz the newly-published v1.7.0 has that PR merged in. Release notes here: https://github.com/localytics/lambda-slack-router/releases/tag/v1.7.0.
